# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditSubActionHandler.cpp
# Commit: 83f3bd0b26e2
# Full Hash: 83f3bd0b26e203bedcae8b14ab5995633b4b3275
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-08-25 09:54:00
# Regressor Bug: 1727185
# File Overlap Count: 1
# Description:
#   Bug 1727185 - Add `MOZ_DIAGNOSTIC_ASSERT` to the constructors of `DeleteNodeTransaction` and `SplitNodeTransaction` and make their users return error before hitting it r=m_kato
#   
#   On Thunderbird, we still have a bug to delete `<html>` and `<body>` elements
#   (bug 1727201).  However, it's hard to know where deletes the unexpected elements
#   from warning messages in the log.  Additionally, it's really serious bug
# ==============================================================================

diff -r 8868fe82dafd -r 83f3bd0b26e2 editor/libeditor/HTMLEditSubActionHandler.cpp
--- a/editor/libeditor/HTMLEditSubActionHandler.cpp	Wed Aug 25 00:21:01 2021 +0000
+++ b/editor/libeditor/HTMLEditSubActionHandler.cpp	Wed Aug 25 00:39:41 2021 +0000
@@ -892,7 +892,7 @@
     return NS_ERROR_EDITOR_DESTROYED;
   }
   NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
-                       "EditorBase::DeleteNodeWithTransaction() failed");
+                       "HTMLEditor::DeleteNodeWithTransaction() failed");
   return rv;
 }
 